packages/meditrak-server/src/routes/EditHandler/EditHandler.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { respond, ObjectValidator, constructRecordExistsWithId } from '@tupaia/utils';
+import { CRUDHandler } from '../CRUDHandler';
+
+/**
+ * Responds to PUT requests for a resource
+ */
+
+export class EditHandler extends CRUDHandler {
+  constructor(req, res) {
+    super(req, res);
+    this.updatedFields = this.req.body;
+  }
+
+  async handleRequest() {
+    await this.validate();
+    await this.editRecord();
+    respond(this.res, { message: `Successfully updated ${this.resource}` });
+  }
+
+  async validate() {
+    // dashboardReport, mapOverlay and dashboardGroup use different id formats and are the only endpoints
+    // which need to overwrite the validation functionality
+    // TODO remove when this task is done https://github.com/beyondessential/tupaia-backlog/issues/723
+    this.validateRecordExists();
```

Comment:
```suggestion
    await this.validateRecordExists();
```
(which will wait for the promise to resolve, and then return undefined)
or
```suggestion
    return this.validateRecordExists();
```
(which will return the promise to be awaited by the caller, which will eventually resolve to the return value of validateRecordExists)

or even 
```suggestion
    return await this.validateRecordExists();
```
(which will wait for the promise to resolve, and then return the result of validateRecordExists)

Reply:
I've updated all of these to `return`